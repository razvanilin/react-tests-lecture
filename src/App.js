import React, { Component } from "react";
import { Route, BrowserRouter as Router } from "react-router-dom";
import Welcome from "./Welcome/Welcome";
import modelInstance from "./data/DinnerModel";
import SelectDish from "./SelectDish/SelectDish";
import Overview from "./Overview/Overview";
import Printout from "./Printout/Printout";
import DishDetail from "./DishDetail/DishDetail";
import "./App.css";

class App extends Component {
  constructor(props) {
    super(props);
    this.state = {
      title: "Dinner Planner"
    };
  }

  render() {
    return (
      <div className="App">
        <Router>
          <header className="App-header">
            <div className="container-fluid" id="headerView">
              <div className="row p-3 mb-2 bg-secondary justify-content-center">
                <h1 className="App-title">{this.state.title}</h1>
              </div>
            </div>
            {/* We rended diffrent component based on the path */}
            <Route exact path="/" component={Welcome} />
            <Route
              path="/search"
              render={() => <SelectDish model={modelInstance} />}
            />
            <Route
              path="/dishDetail/:dishId"
              render={(props) => <DishDetail model={modelInstance} info={props}/>}
            />
            <Route
              path="/overview"
              render={() => <Overview model={modelInstance} />}
            />
            <Route
              path="/printout"
              render={() => <Printout model={modelInstance} />}
            />
          </header>
        </Router>
      </div>
    );
  }
}

export default App;
