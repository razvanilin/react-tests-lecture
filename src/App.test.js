import React from "react";
import renderer from "react-test-renderer";
import { MemoryRouter } from "react-router-dom";

import App from "./App";

it("renders correctly", () => {
  const tree = renderer
    .create(
      <MemoryRouter>
        <App />
      </MemoryRouter>
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});
