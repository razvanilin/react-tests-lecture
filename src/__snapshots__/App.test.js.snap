// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="App"
>
  <header
    className="App-header"
  >
    <div
      className="container-fluid"
      id="headerView"
    >
      <div
        className="row p-3 mb-2 bg-secondary justify-content-center"
      >
        <h1
          className="App-title"
        >
          Dinner Planner
        </h1>
      </div>
    </div>
    <div
      className="Welcome"
    >
      <div
        className="row"
      >
        <div
          className="col"
        />
        <div
          className="col-5 text-center"
        >
          <div
            className="welcomeText"
          >
            Lorem ipsum dolor sit amet, an mei impetus dissentiunt. Reque mundi no eos, sea lorem eripuit pericula ad, mel aperiam voluptatum in. Mei suas verear instructior ea. Eu has incorrupte instructior.
          </div>
          <div
            className="welcomeButton"
          >
            <a
              href="/search"
              onClick={[Function]}
            >
              <button
                className="btn btn-warning"
                type="button"
              >
                Create new dinner
              </button>
            </a>
          </div>
        </div>
        <div
          className="col"
        />
      </div>
    </div>
  </header>
</div>
`;
